test/Microsoft.AspNet.Razor.Runtime.Test/XmlDocumentationProviderTest.cs

```diff
@@ -0,0 +1,173 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#if !DNXCORE50
+using System;
+using System.Reflection;
+using Microsoft.AspNet.Razor.Runtime.TagHelpers;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    public class XmlDocumentationProviderTest
+    {
+        private static readonly TypeInfo DocumentedTagHelperTypeInfo = typeof(DocumentedTagHelper).GetTypeInfo();
+        private static readonly PropertyInfo DocumentedTagHelperSummaryPropertyInfo =
+            DocumentedTagHelperTypeInfo.GetProperty(nameof(DocumentedTagHelper.SummaryProperty));
+        private static readonly PropertyInfo DocumentedTagHelperRemarksPropertyInfo =
+            DocumentedTagHelperTypeInfo.GetProperty(nameof(DocumentedTagHelper.RemarksProperty));
+        private static readonly PropertyInfo DocumentedTagHelperRemarksSummaryPropertyInfo =
+            DocumentedTagHelperTypeInfo.GetProperty(nameof(DocumentedTagHelper.RemarksAndSummaryProperty));
+
+        [Fact]
+        public void CanReadXml()
+        {
+            // Act. Ensuring that reading the Xml file doesn't throw.
+            var xmlDocumentationProvider = new XmlDocumentationProvider(DocumentedTagHelper.XmlTestFileLocation);
+        }
+
+        public static TheoryData GetSummaryData
+        {
+            get
+            {
+                // id, expectedSummary
+                return new TheoryData<string, string>
+                {
+                    {
+                        XmlDocumentationProvider.GetId(DocumentedTagHelperTypeInfo),
+                        "The summary for the type."
```

Comment:
Rinse and repeat comment in `TagHelperUsageDescriptorFactoryTest` about common `<summary>` and `<remarks>` content.